pkg/apis/storage/validation/validation_test.go

```diff
@@ -1867,11 +1869,26 @@ func TestCSIDriverValidationUpdate(t *testing.T) {
 		},
 	}
 
-	// Currently there is only one success case: exactly the same
-	// as the existing object.
-	successCases := []storage.CSIDriver{old}
+	// Currently we compare the object against itself
+	// and ensure updates succeed
+	successCases := []storage.CSIDriver{
+		old,
+		// An invalid FSGroupPolicy should still pass
+		{
+			ObjectMeta: metav1.ObjectMeta{Name: driverName},
+			Spec: storage.CSIDriverSpec{
+				AttachRequired: &attachNotRequired,
+				PodInfoOnMount: &notPodInfoOnMount,
+				VolumeLifecycleModes: []storage.VolumeLifecycleMode{
+					storage.VolumeLifecycleEphemeral,
+					storage.VolumeLifecyclePersistent,
+				},
+				FSGroupPolicy: &invalidFSGroupPolicy,
+			},
+		},
+	}
 	for _, csiDriver := range successCases {
-		if errs := ValidateCSIDriverUpdate(&csiDriver, &old); len(errs) != 0 {
+		if errs := ValidateCSIDriverUpdate(&csiDriver, &csiDriver); len(errs) != 0 {
```

Comment:
of course this passes because we're passing in the exact same two objects. Should we at least pass in two different copies to make sure the validation method is not doing some funky pointer comparisons?

Reply:
I've added a DeepCopy here and am passing these in for comparison now.